### PR TITLE
Bedrock: update version of component-boilerplate and add email to conditional section

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

>  Version of component-boilerplate updated to prevent rare race condition with graphQL queries

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3171
 - Process.st launch checklist: https://app.process.st/runs/DEV-3171-Component-Boilerplate-and-mint-update-o3odNUy7pswnNgLZQ4RPtA

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/bedrock-components/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.2.1] - 2021-11-22
+
+### Fixed
```
Comment (DerekSiemens) [inline packages/bedrock-components/CHANGELOG.md:12]: changelog should be updated for the new condional section changes

File: packages/bedrock-components/package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "generate": "stencil generate"
   },
   "dependencies": {
-    "@saasquatch/component-boilerplate": "^1.2.2",
+    "@saasquatch/component-boilerplate": "^1.2.3-11",
```
Comment (DerekSiemens) [inline packages/bedrock-components/package.json:32]: Flagging pre-release version, update needed before merge and deploy

File: packages/bedrock-components/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/bedrock-components",
-  "version": "1.2.0",
+  "version": "1.2.1-6",
```
Comment (DerekSiemens) [inline packages/bedrock-components/package.json:3]: Probably should just bump to 1.3.0

Comment (DerekSiemens) [inline packages/bedrock-components/package.json:3]:
```suggestion
  "version": "1.3.0",
```

File: packages/bedrock-components/package.json (hunk 2 above)
Comment (DerekSiemens) [inline packages/bedrock-components/package.json:32]:
```suggestion
    "@saasquatch/component-boilerplate": "^1.3.0",
```

File: packages/bedrock-components/src/components/sqb-conditional-section/sqb-conditional-section.feature [hunk 4]
```diff
@@ -3,41 +3,51 @@
...
     When the widget is loaded
-    Then "Hello" will not be shown
+    Then "Hello" is not shown
     When the user's custom field "foo" is set to true
-    Then "Hello" will be shown
+    Then "Hello" is shown
+
+  @motivating
+  Scenario: Content can be hidden or shown depending on user's email
+    Given the component's condition prop is set to "$not('example.com' in $substringAfter(user.email, '@'))"
```
Comment (loganvolkers) [inline packages/bedrock-components/src/components/sqb-conditional-section/sqb-conditional-section.feature:30]: Would have been convenient to include `emailDomain` as a thing separate from `email` if we're getting customers to build logic on it.

```suggestion
    Given the component's condition prop is set to "$not('example.com' = user.emailDomain)"
```

**Review by loganvolkers — DISMISSED** [review]
Looks good to me

**Review by DerekSiemens — APPROVED** [review]

**Pull request merged**